goalgorilla/open_social
# Issue #3316240 by nkoporec: Include the VTIMEZONE component in the ICS file

## Problem
Some calendar apps requires to have the VTIMEZONE component, otherwise the date will have the wrong times in the user calendar. We have noticed that Microsoft Outlook has this problem, where times are not correctly displayed, when you add the VTIMEZONE component to the .ics file, then the time is set correctly in the Outlook calendar view.

## Solution
 Add the VTIMEZONE component to the ICS file.

## Issue tracker
https://www.drupal.org/project/social/issues/3316240

## Theme issue tracker
*[Required if applicable] Paste a link to the drupal.org theme issue queue item, either from [socialbase](https://www.drupal.org/project/socialbase) or [socialblue](https://www.drupal.org/project/socialblue). If any other issue trackers were used, include links to those too.*

## How to test
1. Enable the social_event_addtocal module
2. Go to /admin/config/opensocial/event-addtocal and enable the add to event calendar buttons
3. Create an event
4. Click the 'Add to event' button and select the ICS format.
5. Check if the file contains the VTIMEZONE component.

## Definition of done
### Before merge

### After merge

## Screenshots
*[Required if new feature, and if applicable] If this Pull Request makes visual changes then please include some screenshots that show what has changed here. A before and after screenshot helps the reviewer determine what changes were made.*

## Release notes
Added the VTIMEZONE component to the ICS file export for better date and time display in external calendar apps.

## Change Record
*[Required if applicable] If this Pull Request changes the way that developers should do things or introduces a new API for developers then a change record to document this is needed. Please provide a draft for a change record or a link to an unpublished change record below. Existing change records can be consulted as example. Please provide a draft for a change record or a link to an unpublished change record below. [Existing change records](https://www.drupal.org/list-changes/social) can be consulted as example.*

## Translations
*[Optional]Translatable strings are always extracted from the latest development branch. To ensure translations remain available for platforms running older versions of Open Social the original string should be added to `translations.php` when it's changed or removed.*
